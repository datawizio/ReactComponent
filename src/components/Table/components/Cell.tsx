import * as React from "react";
import { TableContext } from "../context";

const Cell: React.FC = props => {
  const { children, ...restProps } = props;
  const {
    tableState: { columnsWidth }
  } = React.useContext(TableContext);

  const style = React.useMemo(() => {
    const output: React.CSSProperties = {};

    const column = children[1]?.props.column;
    const columnKey = column ? column.key : "";

    if (columnKey) {
      output.width = `${columnsWidth[columnKey]}px`;
      return output;
    }
  }, [children, columnsWidth]);

  return (
    <td {...restProps} style={style}>
      <div className="dw-table__cell">{children}</div>
    </td>
  );
};

export default Cell;
