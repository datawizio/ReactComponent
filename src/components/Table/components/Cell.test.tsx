import React from "react";
import { shallow } from "enzyme";
import Cell from "./Cell";

const setUp = (props?) => shallow(<Cell {...props} />);

xdescribe("Cell component", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("rendered Cell corectly", () => {
    expect(component).toMatchSnapshot();
  });

  it("Cell has 1 cell class ", () => {
    const cellClass = component.find(".dw-table__cell");
    expect(cellClass.length).toBe(1);
  });
});
