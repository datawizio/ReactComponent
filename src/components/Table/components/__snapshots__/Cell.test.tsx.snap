// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component rendered Cell corectly 1`] = `undefined`;
